be/src/olap/rowset/segment_v2/column_reader.h

```diff
@@ -187,6 +196,25 @@ class ColumnReader {
     std::unique_ptr<BloomFilterIndexReader> _bloom_filter_index;
 };
 
+class ArrayColumnReader : public ColumnReader {
```

Comment:
It's a bad class design to make ArrayColumnReader directly inheriting from ColumnReader. You'd better start with a ColumnReader interface and design ScalarColumnReader and ArrayColumnReader separately.